cylc/flow/prerequisite.py

```diff
@@ -102,6 +104,12 @@ class Prerequisite:
     SATISFIED_TEMPLATE = 'bool(self._satisfied[("%s", "%s", "%s")])'
     MESSAGE_TEMPLATE = r'%s/%s %s'
 
+    DEP_STATE_SATISFIED = 'satisfied naturally'
+    DEP_STATE_SATISFIED_BY = 'satisfied by {} mode'
+    DEP_STATE_OVERRIDDEN = 'force satisfied'
+    DEP_STATE_UNSATISFIED = False
+    SATISFIED_MODE_RE = re.compile(r'satisfied by .* mode')
```

Comment:
I took these out in #6293 and replaced with the `SatisfiedState` type above and rely on type checking to flag any erroneous/unexpected values inserted in the `self._satisfied` dict. (For example there was/is no class attribute for `'satisfied from database'` which was used elsewhere)

Is there a real need to record if a prerequisite was satisfied by a particular run mode?